### PR TITLE
Renamed Validator implementation for Operator and Package records

### PR DESCRIPTION
**Pull request by calvinrp**
Although it is a validator (and implements `Validator` trait), it is used to reflect the state of the log. Renamed to `OperatorState` and `PackageState`. See issue: https://github.com/bytecodealliance/registry/issues/94

### CODE REVIEW

**Comment by calvinrp** [comment]
There are ~7 files that import both into the same namespace. I'm on board with those suggestions.

**Comment by Kylebrown9** [comment]
I guess for change 2, the logical place might be to just make them all available in the `protocol` crate root. I don't know that we have a better place and creating some new module e.g. `mod log { pub use ... }` in the root or adding a new level to the hierarchy seems a bit silly.

File: crates/protocol/src/package/mod.rs [hunk 1]
```diff
@@ -8,7 +8,7 @@ mod model;
 mod validate;
 
 pub use model::{PackageEntry, PackageRecord, Permission};
-pub use validate::{Head, PackageState, Release, ReleaseState, ValidationError};
```
Comment (Kylebrown9) [inline crates/protocol/src/package/mod.rs]: There's still an inconsistency here between `PackageState`/`PackageValidationError` and `Release`/`ReleaseState`, which I think is a good focal point for the main question we have to answer.

Which of the following do we want in the codebase?
1. all names get log-name prefixing including `PackageRelease` & `PackageReleaseState`;
2. some names get log-name prefixing, but others like `Release` and `ReleaseState` don't;
3. no names are log-name prefixed and we consistently use `package::` and `operator::` for disambiguation like we do today.

I'm not a fan of the inconsistency of 2, without trying them both I don't have a huge preference between 1 & 3. What do other people think?

@lann @peterhuene

Comment (lann) [reply crates/protocol/src/package/mod.rs]: I'm fine with 1 or 3. I would probably pick 1 for this particular case.

Comment (calvinrp) [reply crates/protocol/src/package/mod.rs]: Option 1 pushed.

**Comment by calvinrp** [comment]
Looks like I have some merge issues that still need fixing

**Comment by calvinrp** [comment]
Going to scrap this one and start over

**Pull request closed without merging**